lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +103,189 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
-                  + " pods by using the `kubectl rollout restart deployment {}`"
-                  + " command to unpase all pods.",
-              target.getDeployment().getMetadata().getName());
-          return;
+          throw e;
         }
       }
     }
   }
 
+  @VisibleForTesting
+  TargetSnapshot getTarget() throws PauserException {
+    return targetSelector.select();
+  }
+
+  @VisibleForTesting
   RequestCoordinator getRequestCoordinator(TargetSnapshot target) {
     return new RequestCoordinator(
         target.getPods().stream()
             .map(p -> new InetSocketAddress(p.getStatus().getPodIP(), target.getAdminPort()))
             .collect(Collectors.toList()));
   }
+
+  @VisibleForTesting
+  PausedDuration pauseInternal(
+      RequestCoordinator requestCoordinator, int pauseDuration, @Nullable Long maxPauseWaitTime) {
+    requestCoordinator.pause(true, maxPauseWaitTime);
+    Instant startTime = Instant.now();
+    Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
+    Instant endTime = Instant.now();
+    return new PausedDuration(startTime, endTime);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  StatusUnmatchedException targetStatusEquals(TargetSnapshot before, TargetSnapshot after) {
+    if (before.getStatus().equals(after.getStatus())) {
+      return null;
+    } else {
+      return new StatusUnmatchedException(STATUS_UNMATCHED_ERROR_MESSAGE);
+    }
+  }
+
+  @Nullable
+  @VisibleForTesting
+  PauserException buildException(
+      @Nullable UnpauseFailedException unpauseFailedException,
+      @Nullable PauseFailedException pauseFailedException,
+      @Nullable GetTargetAfterPauseFailedException getTargetAfterPauseFailedException,
+      @Nullable StatusCheckFailedException statusCheckFailedException,
+      @Nullable StatusUnmatchedException statusUnmatchedException) {
+    PauserException pauserException = null;
+
+    // Unpause issue is the most critical because it might cause system down.
+    if (unpauseFailedException != null) {
+      pauserException = unpauseFailedException;
+    }
+
+    // Pause Failed is second priority because pause issue might be caused by configuration error.
+    if (pauseFailedException != null) {
+      if (pauserException == null) {
+        pauserException = pauseFailedException;
+      } else {
+        pauserException.addSuppressed(pauseFailedException);
+      }
+    }
+
+    // Get target issue is third priority because this issue might be caused by temporary issue, for
+    // example, network issues.
+    if (getTargetAfterPauseFailedException != null) {
+      if (pauserException == null) {
+        pauserException = getTargetAfterPauseFailedException;
+      } else {
+        pauserException.addSuppressed(getTargetAfterPauseFailedException);
+      }
+    }
+
+    // Status check issue is third priority because this issue might be caused by temporary issue,
+    // for example, targetAfterPause is null.
```

Comment:
What causes `targetAfterPause is null`?

Reply:
Ah, sorry. This comment does not have meaningful information.

For example, if we cannot get the pod or deployment information by using the Kubernetes API, the `targetAfterPause` will be `null`, and enter this conditional branch.

I will fix this comment. Thank you for your question!

Reply:
Fixed in 639e132e4766b0299b8b91047b8b633f0371072f and b93c054f00233d3d37ebb1571c9031f6d68e7747.